extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -513,8 +526,8 @@ public static Table readPartitionedTableInferSchema(
                     getUnboxedTypeIfBoxed(partitionValue.getClass()), null, ColumnDefinition.ColumnType.Partitioning));
         }
         allColumns.addAll(schemaInfo.getFirst());
-        return readPartitionedTable(recordingLocationKeyFinder, schemaInfo.getSecond(),
-                TableDefinition.of(allColumns));
+        return readPartitionedTable(readInstructions.isRefreshing() ? locationKeyFinder : initialKeys,
```

Comment:
This is a little unfortunate if it causes us to scan for keys an extra time initially. I suppose it depends on the key finder implementation from a performance standpoint.

Reply:
I think some of these interfaces could be improved. For example, I think there is room for the user to explicitly provide the schema instead of inferring it from the first file. 

This situation could also be improved by updating or changing TableLocationKeyFinder - we shouldn't need to read all of the table locations if we only want just the first one (for inferring schema).

Reply:
I think we already have an exposed public method that allows the user to specify. I also think it's not that easy for a user to do.

The key finders were built for that static use case. I agree that we could have a richer interface (`findFirstKey`, `findNewKeys(Set)`).